aws/resource_aws_kinesis_analytics_application.go

```diff
@@ -614,6 +614,13 @@ func resourceAwsKinesisAnalyticsApplicationCreate(d *schema.ResourceData, meta i
 		d.SetId(aws.StringValue(output.ApplicationSummary.ApplicationARN))
 		return nil
 	})
+
+	if isResourceTimeoutError(err) {
+		var output *kinesisanalytics.CreateApplicationOutput
```

Comment:
Nit: The updated code could simplify this handling by having this variable declaration before the original `resource.Retry()` and calling `d.SetId()` after all errors have been checked, e.g.

```go
var output *kinesisanalytics.CreateApplicationOutput
err := resource.Retry(1*time.Minute, func() *resource.RetryError {
	var err error
	output, err = conn.CreateApplication(createOpts)
	if err != nil {
		/* ... existing error handling ... */
	}
	return nil
}
if isResourceTimeoutError(err) {
	output, err = conn.CreateApplication(createOpts)
}
if err != nil {
	return fmt.Errorf("error creating Kinesis Analytics Application: %s", err)
}
d.SetId(aws.StringValue(output.ApplicationSummary.ApplicationARN))
```

Reply:
Gotcha! Will keep an eye out for this in future files!